frontends/queues/queues/compile_queue.py

```diff
@@ -0,0 +1,96 @@
+# pylint: disable=import-error
+import sys
+import calyx.builder as cb
+import queues.queue_call as qc
+import queues.strict_or_rr as strict_or_rr
+import queues.fifo as fifo
+import queues.flow_inference as fi
+import json
+import os
+
+rr_id = 0
+strict_id = 0
+
+
+def create(data, lower, upper, prog, fifo_queue):
+    """
+    Recursively creates the PIFO by crawling the json file `data`, which represents
+    a Rio program. `lower' and `upper` are the bounds for which a flow is restricted
+    to.
+    """
+    global rr_id
+    global strict_id
+    if isinstance(data, dict):
```

Comment:
Should this maybe be changed into an assert? We need the data to be a dict, or else we do nothing clever. Right?